src/rgw/rgw_common.h

```diff
@@ -993,78 +993,160 @@ inline ostream& operator<<(ostream& out, const rgw_obj_key &o) {
   }
 }
 
+/* Holds info on whether the request should be
+ * validated via EC2 signature or Auth tokens.
+ * Holds value when the action is COPY
+ * Holds value when the token to be validated is from a presigned URL */
+
+class authorization_method {
+  private:
+    bool _token_validation;
+    bool _copy_action;
+    bool _url_type_token;
+    bool _acl_main_override;
+    bool _acl_copy_override;
+    string _token;
+    string _copy_source;
+
+  public:
+    inline bool get_token_validation()
+    {
+      return _token_validation;
+    }
+    inline void set_token_validation(bool method)
+    {
+      _token_validation = method;
+    }
+    inline string get_token()
+    {
+      return _token;
+    }
+    inline void set_token(string tok)
+    {
+      _token = tok;
+    }
+    inline bool get_copy_action()
+    {
+      return _copy_action;
+    }
+    inline void set_copy_action(bool action)
+    {
+      _copy_action = action;
+    }
+    inline string get_copy_source()
+    {
+      return _copy_source;
+    }
+    inline void set_copy_source(string source)
+    {
+      _copy_source = source;
+    }
+    inline bool get_url_type_token()
+    {
+      return _url_type_token;
+    }
+    inline void set_url_type_token(bool val)
+    {
+      _url_type_token = val;
+    }
+    inline bool get_acl_main_override()
+    {
+      return _acl_main_override;
+    }
+    inline void set_acl_main_override(bool val)
+    {
+      _acl_main_override = val;
+    }
+    inline bool get_acl_copy_override()
+    {
+      return _acl_copy_override;
+    }
+    inline void set_acl_copy_override(bool val)
+    {
+      _acl_copy_override = val;
+    }
+
+
+    authorization_method(bool method, bool action, bool url_token) :
+      _token_validation(method),
+      _copy_action(action),
+      _url_type_token(url_token) { }
```

Comment:
Minor: I forgot to add acl_override bools here for initialization. Maybe we should put them in constructor.